### PR TITLE
Milestone 3

### PR DESCRIPTION
**Pull request by jfischoff**
This completes the DAO contract system. It adds in the general and trip proposal types and configuration. It also improves the performance of tallying.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -98,6 +116,45 @@ If the vote is valid, a UTxO is created with the a vote datum that stores the pr
 
 Users have to include min Ada for the their vote NFT, and additionally ada to cover the fees of tallying their vote. Users can cancel their votes at any time and receive all the assets the used to create the vote UTxO.
```
Comment (Randy1Burrell) [inline README.md:117]: > Users can cancel their votes at any time and receive all the assets the used to create the vote UTxO.

Should be 
> Users can cancel their votes at any time and receive all the assets they used to create the vote UTxO.

File: scripts/generate-datums.sh [hunk 2]
```diff
@@ -29,11 +29,16 @@ voteCurrencySymbol=$(cat $thisDir/vote-minter-policy-id.txt)
 voteTokenName=564F5445
 voteValidator=$(cat $thisDir/vote-validator-hash.txt)
 updateMajoriyPercent=500 # 50 percent
-upgradeRelativeMajorityPercent=110 # ten percent
+upgradeRelativeMajorityPercent=110 # eleven percent
+genericMajoriyPercent=500 # 50 percent
```
Comment (Randy1Burrell) [inline scripts/generate-datums.sh]: Was the t left out of `genericMajoriyPercent` deliberately?

File: scripts/generate-datums.sh [hunk 3]
```diff
@@ -29,11 +29,16 @@ voteCurrencySymbol=$(cat $thisDir/vote-minter-policy-id.txt)
...
+genericRelativeMajorityPercent=110 # eleven percent
+travelMajoriyPercent=500 # 50 percent
```
Comment (Randy1Burrell) [inline scripts/generate-datums.sh:35]: Same as above

File: src/Canonical/Treasury.hs [hunk 4]
```diff
@@ -73,22 +69,74 @@ data TreasuryValidatorConfig = TreasuryValidatorConfig
...
+
+ownValueAndValidator :: [TreasuryTxInInfo] -> TxOutRef -> (Value, ValidatorHash)
+ownValueAndValidator ins txOutRef = go ins where
+  go = \case
+    [] -> traceError "The impossible happened"
+    TreasuryTxInInfo {tTxInInfoOutRef, tTxInInfoResolved = TreasuryTxOut{tTxOutAddress = Address {..}, ..}} :xs ->
+      if tTxInInfoOutRef == txOutRef then
+        case addressCredential of
+          ScriptCredential vh -> (tTxOutValue, vh)
+          _ -> traceError "Impossible. Expected ScriptCredential"
```
Comment (Randy1Burrell) [inline src/Canonical/Treasury.hs:116]: In the case of debugging, I think these error messages should be a bit more informative. Would an average user understand what `ScriptCredential` is?

**Review by Randy1Burrell — DISMISSED** [review]
@jfischoff great work. I left a few comments but otherwise, this looks good.

File: scripts/generate-datums.sh (hunk 2 above)
Comment (jfischoff) [reply scripts/generate-datums.sh]: No it's a typo

File: src/Canonical/Treasury.hs (hunk 4 above)
Comment (jfischoff) [reply src/Canonical/Treasury.hs:116]: This code path cannot be executed. The message is not for a user but to make the code compile.

**Review by Randy1Burrell — APPROVED** [review]

**Pull request merged**